src/shell/shell_cmds.cpp

```diff
@@ -1545,22 +1545,18 @@ void DOS_Shell::CMD_PATH(char *args){
 	}
 }
 
-void DOS_Shell::CMD_VER(char *args) {
+bool set_ver(const char *word, const char *args);
```

Comment:
Please don't do forward declarations of functions this way. This a new function in `dos` module, therefore should be declared in `dos_inc.h` and be named something like `DOS_ParseVersion`.

Also, I would much prefer if it was more like a pure function returning `DOS_Version` struct instead of modifying global variables, so it could be used this way:
```C++
const auto new_version = DOS_ParseVersion(args);
if (<new version is ok>)
        dos.version = new_version;
else
        WriteOut(…);
```

Reply:
I have already declared it in dos_inc.h and name it DOS_ParseVersion. A new struct will be used to indicate the major and minor version and whether the setting succeeds.

Reply:
Why not existing one? If the version is incorrect you can return 0-filled struct.

Reply:
Good suggestion. I have modified the code to treat the version as invalid if and only if both major and minor are set to 0. Please check out the current code.